stl/inc/xstring

```diff
@@ -2734,7 +2864,7 @@ private:
         const size_type _Right_size   = _Right_data._Mysize;
         const _Elem* const _Right_ptr = _Right_data._Myptr();
         auto& _My_data                = _Mypair._Myval2;
-        if (_Right_size < _BUF_SIZE) { // stay small, don't allocate
+        if (!_Right_data._Large_string_engaged()) { // stay small, don't allocate

```

Comment:
I believe this one is correct, one of the few places I actually did change runtime behavior. @mnatsuhara